### PR TITLE
Add concave hull knn and update the name of original concave hull

### LINKED ISSUE #3018: [feature] Adapted k-neighbour concave hull plugin as QGIS 3 algorithm
Original commit: https://github.com/qgis/QGIS/commit/52ccc43ebb238db84d2ef4758d864a4257848a41 by rudivs

Co-authored-by: Detlev Neumann <dneumann@geospatial-services.de>

### PR DESCRIPTION
**Pull request by havatv**
A new concave hull algorithm has been added.  It is now documented, and the title of the original concave hull algorithm is changed (Fixes #3018).

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -483,6 +483,38 @@ See also
 :ref:`qgisconvexhull`
 
 
+.. _qgisconcavehullknn:
+
+Concave hull (k-nearest neighbor)
+---------------------------------
+Computes the concave hull of the features in an input point layer
+using nearest neighbors.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: How about picking more details in the gui help (for both the description and the parameters)? eg info about vertices usage
The doc is rather light in its current shape imho...

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -483,6 +483,38 @@ See also
...
+
+Parameters
+..........
+``Input point layer`` [vector: point]
+  Point vector layer to calculate the concave hull.
+
+``Number of neighboring points to consider`` [number]
+  Number (integer) from 3 (maximum concave hull) to 999999999 (convex hull).
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: mainly inspired by the GUI: `Determines the concaveness of the output polygon. A lower number will result in a concave hull that follows the points very closely, while a higher number will tend to a convex hull.`

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -483,6 +483,38 @@ See also
...
+
+  Default: *3*
+
+``Field`` [tablefield: any]
+  To create concave hulls by class (one concave hull per class).
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: In the ideal world, should be in a conjugued form (?) ie `creates...`, or... from GUI, `If a field is selected, generates individual polygon in the output layer for each group of unique values in the field.` (something like that)

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 4]
```diff
@@ -483,6 +483,38 @@ See also
...
+.. _qgisconcavehullknn:
+
+Concave hull (k-nearest neighbor)
+---------------------------------
+Computes the concave hull of the features in an input point layer
+using nearest neighbors.
+
+Parameters
+..........
+``Input point layer`` [vector: point]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Given the description in the GUI, it's not only about point. Also takes line or polygon (reason why mentioning how these geometries are handled is necessary in the description section).

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 5]
```diff
@@ -483,6 +483,38 @@ See also
 :ref:`qgisconvexhull`
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Done.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 6]
```diff
@@ -483,6 +483,38 @@ See also
...
 
 
+.. _qgisconcavehullknn:
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Fixed.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 7]
```diff
@@ -483,6 +483,38 @@ See also
...
+
+Concave hull (k-nearest neighbor)
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Done.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Good idea. Done.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Thanks .  Done.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 2 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Fixed.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 8]
```diff
@@ -483,6 +483,38 @@ See also
...
+..........
+``Input point layer`` [vector: point]
+  Point vector layer to calculate the concave hull.
+
+``Number of neighboring points to consider`` [number]
+  Number (integer) from 3 (maximum concave hull) to 999999999 (convex hull).
+
+  Default: *3*
+
+``Field`` [tablefield: any]
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Fixed.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 3 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I have tried a new formulation.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 9]
```diff
@@ -480,7 +480,57 @@ Outputs
...
+:ref:`qgisconvexhull`, :ref:`qgisknearestconcavehull`
+
+
+.. _qgisknearestconcavehull:
+
+Concave hull (k-nearest neighbor) |34| 
+--------------------------------------
+This algorithm generates a concave hull polygon from a set of points.
+If the input layer is a line or polygon layer, it will use the
+vertices / nodes.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I'm really not sure about the subtle difference between both, but wouldn't one suffice? My vote would go to `vertices`.
```suggestion
vertices.
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 10]
```diff
@@ -480,7 +480,57 @@ Outputs
...
+
+The number of neighbours to consider determines the concaveness of the
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: en-US
```suggestion
The number of neighbors to consider determines the concaveness of the
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 11]
```diff
@@ -480,7 +480,57 @@ Outputs
...
+output polygon.
+A lower number will result in a concave hull that follows the points very
+closely, while a higher number will have a smoother shape.
+The minimum number of neighbour points to consider is 3.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
The minimum number of neighbor points to consider is 3.
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 12]
```diff
@@ -480,7 +480,57 @@ Outputs
...
+input layer using unique values in that field and generate individual
+polygons in the output layer for each group. 
+
+Parameters
+..........
+``Input layer`` [vector: any]
+  Vector layer to calculate the concave hull.
+
+``Number of neighboring points to consider`` [number]
+  Number (integer) from 3 (maximum concave hull) to 999999999 (convex
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: My suggestion aimed to remove that line... Opinion (and see next)?

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 13]
```diff
@@ -480,7 +480,57 @@ Outputs
...
+  hull).
+  Determines the concaveness of the output polygon.
+  A lower number will result in a concave hull that follows the points
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: And to keep details from above, we can add `A lower number (minimum: 3) will result in...`

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 14]
```diff
@@ -480,7 +480,57 @@ Outputs
...
+  very closely, while a higher number will tend to a convex hull.
+
+  Default: *3*
+
+``Field`` [tablefield: any] Optional
+  If specified, one concave hull polygon is generated for each unique
+  value of the field (by selecting points using this value).
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: should we use features (because of lines and polygons) instead of points?

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Agree.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 12 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Will remove it.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 13 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I have another suggestion.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 15]
```diff
@@ -480,7 +480,58 @@ Outputs
...
+  Determines the concaveness of the output polygon.
+  A small number will result in a concave hull that follows
+  the points very closely, while a high number will make
+  the polygon look more like the convex hull (if the number
+  is equal to or larger than the number of features, the
+  result will be the convex hull).
+
+  Default (and minimum): *3*
+
+``Field`` [tablefield: any] Optional
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Should be on the next line. 
```suggestion
``Field`` [tablefield: any]
  Optional

```

**Pull request merged**